qutebrowser/mainwindow/statusbar/bar.py

```diff
@@ -287,6 +287,8 @@ def _clear_widgets(self):
                        self.backforward, self.tabindex,
                        self.keystring, self.prog, self.clock, *self._text_widgets]:
             assert isinstance(widget, QWidget)
+            if widget in [self.prog, self.backforward]:
+                widget.enabled=False
```

Comment:
For the mypy error probably just slap a `# type: ignore[attr-defined]` on the end of the line.
I think the whole statusbar/ folder is a bit beyond consistent typing and polymorphism at this point an needs a good refactor.

Reply:
I've done exactly that.

Reply:
Actually that's a goof on my part, i should've used the .setEnabled method instead of setting the variable directly. 
Seems to work fine with mypy then.

Reply:
Have you tested the behaviour of this change? I don't think `widget.enabled = False` is equivalent to `widget.setEnabled(False)` in this case.
Possibly the widget implementations should be using the more generic QWidget stuff but I think the `enabled` attribute is just added in locally to the subclasses. So `.setEnabled()` has no effect on it.

Reply:
i think i was testing this on the old branch thinking it was the new one somehow lol. You are totally right.

Reply:
I reverted that last commit

Reply:
Your branch now seems to contain *only* that commit.

Reply:
Unless I'm seeing something wrong that doesn't seem to be the case
this is the only code change that i see in this pr:
![image](https://user-images.githubusercontent.com/9768508/190142447-2b1b392d-0d23-4abf-b1e5-9c6c4d2b202e.png)
and the branch contains the 3 original commits without that errant setEnabled() one.

Reply:
Ah yes, my bad - I misunderstood the conversion above and didn't have a closer look at the change myself yet. Nevermind me (for now) then!

Reply:
No problem!